### PR TITLE
[#159556607] Validate Assign Asset

### PR DESCRIPTION
**Pull request by HawiCaesar**
## What does this PR do?
Disable `Assign Asset` button when no user is assigned to asset. This is to prevent the user from clicking `Assign Asset` without filling in a user.
## Description of Task to be completed?
- add button state to `Assign Asset` button
- change state from disabled when user selected in dropdown
- change css for button and refactor css to allow for reuse.
## How should this be manually tested?
`npm test` OR
- Login to ART
- Go to the assets page
- click on asset (one that is unassigned)
- you will see the `Assign Asset` button is disabled
- when selecting a user, the `Assign Asset` becomes clickable
## What are the relevant pivotal tracker stories?
[159556607](https://www.pivotaltracker.com/story/show/159556607)
## Any background context you want to add?
n/a
## Important notes
n/a
## Packages installed
n/a
## Deployment note
n/a
## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()
## Todos
- [x] Raise PR
- [x] Test
- [ ] Documentation
## Screenshots
screen shot 2018-08-13 at 17 21 13

### CODE REVIEW

File: src/_css/sharedStyling.scss [hunk 1]
```diff
@@ -43,7 +43,7 @@ input,
   font-style: normal;
 }
 
-#blue-rounded-button {
+@mixin blue-rounded-buttons {
```
Comment (sachungo) [inline src/_css/sharedStyling.scss]: nice 👍

File: src/components/common/ButtonComponent.jsx [hunk 2]
```diff
@@ -3,17 +3,29 @@ import { Button } from 'semantic-ui-react';
 import PropTypes from 'prop-types';
 
 const ButtonComponent = ({
-  buttonName, color, handleClick, buttonState, customCss
+  buttonName, color, handleClick, buttonState, customCss, disabledState
 }) => {
   if (color === 'primary') {
     return (
-      <Button primary onClick={handleClick} loading={buttonState} className={customCss}>
+      <Button
```
Comment (sachungo) [inline src/components/common/ButtonComponent.jsx]: I recommend a minor change to this component. The major differences between the props for primary and secondary color are:
- primary
- type
- loading

Therefore, we can have only one return. Something like:
```
const PRIMARY_COLOR = 'primary';

const ButtonComponent = ({
  buttonName,
  color,
  handleClick,
  buttonState,
  customCss,
  disabledState
}) => {
  let additionalProps = {
    secondary: true,
    type: "reset"
  }
  if (color === PRIMARY_COLOR) {
    additionalProps = {
      primary: true,
      loading: buttonState
    };
  }

  return (
    <Button
      {...additionalProps}
      onClick={handleClick}
      disabled={disabledState}
      className={customCss}
    >
      {buttonName}
    </Button>
  );
}
```

File: src/_css/sharedStyling.scss (hunk 1 above)
Comment (HawiCaesar) [reply src/_css/sharedStyling.scss]: Thank you!

File: src/components/common/ButtonComponent.jsx (hunk 2 above)
Comment (HawiCaesar) [reply src/components/common/ButtonComponent.jsx]: Makes sense. 

Done.

**Pull request merged**